Scenarios/AKS-Secure-Baseline-PrivateCluster/Terraform/05-Network-LZ/app-gateway.tf

```diff
@@ -23,33 +60,45 @@ resource "azurerm_subnet_network_security_group_association" "appgwsubnet" {
   network_security_group_id = module.appgw_nsgs.appgw_nsg_id
 }
 
+# based on the structure of the appgws map are deployed multiple appplication gateway, usually this is used in the blue green scenario
 resource "azurerm_public_ip" "appgw" {
-  name                = "appgw-pip"
+  for_each = { for appgws in local.appgws : appgws.name_prefix => appgws if appgws.appgw_turn_on == true}
+  name                = "appgw-pip-${each.value.name_prefix}"
   resource_group_name = azurerm_resource_group.spoke-rg.name
   location            = azurerm_resource_group.spoke-rg.location
   allocation_method   = "Static"
   sku                 = "Standard"
 }
 
+# based on the structure of the appgws map are deployed multiple appplication gateway, usually this is used in the blue green scenario
 module "appgw" {
   source = "./modules/app_gw"
   depends_on = [
     module.appgw_nsgs
   ]
-
+  for_each = { for appgws in local.appgws : appgws.name_prefix => appgws if appgws.appgw_turn_on == true}
   resource_group_name  = azurerm_resource_group.spoke-rg.name
   virtual_network_name = azurerm_virtual_network.vnet.name
   location             = azurerm_resource_group.spoke-rg.location
-  appgw_name           = "lzappgw"
+  appgw_name           = "lzappgw-${each.value.name_prefix}"
   frontend_subnet      = azurerm_subnet.appgw.id
-  appgw_pip            = azurerm_public_ip.appgw.id
+  appgw_pip            = azurerm_public_ip.appgw[each.value.name_prefix].id
 
 }
 
+# the app gateway name for each instance provisioned
 output "gateway_name" {
-  value = module.appgw.gateway_name
+  value = { for appgws in module.appgw : appgws.gateway_name => appgws.gateway_name}
```

Comment:
Please add comment indicating that if Blue/Green scenario is not used then this code can be simplified by removing the for_each look and using a single app gateway.

Reply:
fixed